features/steps/fakeapi.go

```diff
@@ -29,3 +31,13 @@ func NewFakeAPI() *FakeAPI {
 func (f *FakeAPI) Close() {
 	f.fakeHTTP.Close()
 }
+
+func (f *FakeAPI) setJSONResponseForGetPageData(statusCode int) {
+	if statusCode == 200 {
+		f.fakeHTTP.NewHandler().Get("/data?uri=%2Feconomy%2Flatest&lang=en").Reply(statusCode).BodyString(`{"type": "article",
+									"description": {"title": "labour market statistics"}}`)
+	} else {
+		f.fakeHTTP.NewHandler().Get("/data?uri=%2Feconomy%2Flatest&lang=en").Reply(statusCode).BodyString(`{"type": "taxonomy_landing_page",
+									"description": {"title": "Economic output and productivity"}}`)
+	}
+}
```

Comment:
So instead of structuring like this, you could make this more re-usable by making a step that's:

`^get page data request to zebedee for ${uri} is successful and returns a ${pageType} page$`

or 

`^get page data request to zebedee for ${uri} returns a 404$`

This could then pass down into the appropriate functions.

Not a blocker for moving forwards but something to think about / adapt if you've got time.

Reply:
done 👍